test/integration/master/kube_apiserver_test.go

```diff
@@ -494,3 +495,84 @@ func TestReconcilerMasterLeaseMultiMoreMasters(t *testing.T) {
 func TestReconcilerMasterLeaseMultiCombined(t *testing.T) {
 	testReconcilersMasterLease(t, 3, 3)
 }
+
+func TestMultiMasterNodePortAllocation(t *testing.T) {
```

Comment:
this is exactly the scenario described in https://github.com/kubernetes/kubernetes/issues/85894